### PR TITLE
Kelsha Darby - User Story 23

### LINKED ISSUE #26: User Story 23, Adding Item Quantity to Cart
As a visitor
When I have items in my cart
And I visit my cart
Next to each item in my cart
I see a button or link to increment the count of items I want to purchase
I cannot increment the count beyond the item's inventory size

### PR DESCRIPTION
**Pull request by kelshadarby**
## Item quantity incrementation (adding to quantity)
Closes #26 

**_Test_**

_Features Spec_
- Added to `item/index_spec` to test that a user can increment items in their cart but NOT beyond the items inventory

_Model Spec_
- Created a cart model spec to test all methods in the cart model

**_View_**
- Added the `+` link to the quantity section of each item

**_Routes_**
- Added a PATCH route to link to the `cart#update` method

**_Controller_**
- Created the `update` method in the controller
     - This method contains a conditional to check if the params passed in are to add or subtract
     - It also checks if the item about to be added will exceed the items inventory

**_Model_**
- Added a `subtract_item` method to help the conditional in the controller

### CODE REVIEW

File: spec/features/cart/show_spec.rb [hunk 1]
```diff
@@ -55,21 +55,41 @@
...
+          end
+        expect(page).to_not have_content("1")
+        expect(page).to have_content("3")
+        end
+      end
+      it "I cannot increment the count of items beyond the item's inventory size" do
+      visit '/cart'
+
+      within "#item-quantity-#{@tire.id}" do
+        15.times do
```
Comment (fredrondina) [inline spec/features/cart/show_spec.rb:73]: nice job shortening this test here

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -12,6 +12,18 @@ def show
...
+  def update
+    item = Item.find(params[:item_id])
+
+    if params[:quantity] == "add" && (cart.contents[item.id.to_s] + 1) <= item.inventory
+      cart.add_item(params[:item_id].to_s)
+    elsif params[:quanity] == "subtract"
+      cart.subtract_item(params[:item_id].to_s)
+    end
+    
+    redirect_to '/cart'
```
Comment (brian-greeson) [inline app/controllers/cart_controller.rb:24]: You could use a path prefix here

File: spec/models/cart_spec.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
+require "rails_helper"
+
+RSpec.describe Cart, type: :model do
```
Comment (fredrondina) [inline spec/models/cart_spec.rb:3]: great job adding these model tests

**Review by fredrondina — APPROVED** [review]

File: spec/models/cart_spec.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+      cart.add_item(tire.id.to_s)
+      cart.add_item(paper.id.to_s)
+      cart.add_item(pencil.id.to_s)
+
+      expect(cart.contents[tire.id.to_s]).to eq(2)
+      expect(cart.contents[paper.id.to_s]).to eq(1)
+      expect(cart.contents[pencil.id.to_s]).to eq(1)
+    end
+    it "subtract_item" do
```
Comment (brian-greeson) [inline spec/models/cart_spec.rb:23]: Needs a space between these tests

File: spec/models/cart_spec.rb [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      cart.add_item(tire.id.to_s)
+      expect(cart.contents[tire.id.to_s]).to eq(2)
+
+      cart.subtract_item(tire.id.to_s)
+      expect(cart.contents[tire.id.to_s]).to eq(1)
+    end
+    it "total_items" do
+      mike = Merchant.create(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
+      meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
+
```
Comment (brian-greeson) [inline spec/models/cart_spec.rb:38]: These tests look really nice. Just add that space between the `it` blocks.

**Review by brian-greeson — CHANGES_REQUESTED** [review]
Great Job!  Looks really good. just a little formatting on your test file. **see comments**

**Pull request merged**